### ISSUE TITLE
Allow unused vars in function arguments for abstract classes/interfaces

### ISSUE THREAD

**Issue by webstrap** [styladev/stylaguide#11]
`no-unused-vars` = args = none (allow in arguments)

If you create a base class that others should extend

```
function parse(data, settings){
  //override that function to parse your concrete data
}
```

It is helpful for people who want to implement to know what is passed to the function.

Alternative suggestion was to add

```
/**
 * ## parse
 *
 * This function will parse different data objects and you can use settings
 * that are passed as well. You get as arguments 
 * data {Object} which contains the json and 
 * settings {Object} which contains your settings.
 *
 * @return {void} void
function parse(){
  //override that function to parse your concrete data
}
```

**Comment by mousemke** [comment]
http://pics.knoblau.ch/Blueprint.jsx__layoutEngine_widget_2016-11-03_11-17-30.png

**Comment by maciejhirsz** [comment]
A common pattern in languages that do have interfaces is to denote with `_` arguments that should be ignored, there is even a setting for that in eslint:

```js
/*eslint no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]*/

function foo(x, _y) {
    return x + 1;
}
foo();
```

It then makes it explicit that the argument is being ignored while still being present and having full jsdoc with type included.

**Comment by mousemke** [comment]
intreresting.  and a much better solution than the others on the table

**Comment by maciejhirsz** [comment]
Closed by #14
